src/OrchardCore.Modules/OrchardCore.Users/UserMenuShapeTableProvider.cs

```diff
@@ -0,0 +1,31 @@
+using System;
+using OrchardCore.DisplayManagement.Descriptors;
+
+namespace OrchardCore.Users;
+
+public class UserMenuShapeTableProvider : IShapeTableProvider
+{
+    private const string _shapePrefix = "UserMenuItems";
+
+    public void Discover(ShapeTableBuilder builder)
+    {
+        // Describe any shape-type that starts with 'UserMenuItems__'.
+        builder.Describe(_shapePrefix + "__*")
+            .OnDisplaying(context =>
+            {
+                if (String.IsNullOrEmpty(context.Shape.Metadata.DisplayType) || context.Shape.Metadata.DisplayType == "Detail")
+                {
+                    return;
+                }
+
+                // UserMenuItems_{displayType} > UserMenuItems.{displayType}.cshtml.
+                context.Shape.Metadata.Alternates.Add(_shapePrefix + '_' + context.Shape.Metadata.DisplayType);
```

Comment:
Maybe use another const `ShapeAlternatePrefix = $"ShapePrefix{_}"` (or only `AlternatePrefix`).

Then maybe string interpolation `$"{ShapeAlternatePrefix}{DisplayType}"`


Note: For info in some providers we use `EncodeAlternateElement()` on any element as the `DisplayType`, in case the element itself contains any `("-", "__")`

Reply:
okay.. `EncodeAlternateElement` is defined in so many places into the project. I moved that into a public extension and reused it everywhere